### PR TITLE
check if PR is already merged in --merge-pr

### LINKED ISSUE #3501: --merge-pr causes a traceback when the PR is already merged
```
Adding comment to easybuild-easyconfigs issue #11679: 'Going in, thanks @boegel!'
Traceback (most recent call last):
  File "/usr/lib64/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib64/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/mnt/ds918plus/eb/nuc-fedora32/software/EasyBuild/4.3.1/lib/python3.8/site-packages/easybuild/main.py", line 533, in <module>
    main()
  File "/mnt/ds918plus/eb/nuc-fedora32/software/EasyBuild/4.3.1/lib/python3.8/site-packages/easybuild/main.py", line 251, in main
    merge_pr(options.merge_pr)
  File "/mnt/ds918plus/eb/nuc-fedora32/software/EasyBuild/4.3.1/lib/python3.8/site-packages/easybuild/tools/github.py", line 1313, in merge_pr
    github_api_put_request(merge_url, github_user, body=body)
  File "/mnt/ds918plus/eb/nuc-fedora32/software/EasyBuild/4.3.1/lib/python3.8/site-packages/easybuild/tools/github.py", line 281, in github_api_put_request
    status, data = url.put(**kwargs)
  File "/mnt/ds918plus/eb/nuc-fedora32/software/EasyBuild/4.3.1/lib/python3.8/site-packages/easybuild/base/rest.py", line 145, in put
    return self.request(self.PUT, url, json.dumps(body), headers, content_type='application/json')
  File "/mnt/ds918plus/eb/nuc-fedora32/software/EasyBuild/4.3.1/lib/python3.8/site-packages/easybuild/base/rest.py", line 174, in request
    conn = self.get_connection(method, url, body, headers)
  File "/mnt/ds918plus/eb/nuc-fedora32/software/EasyBuild/4.3.1/lib/python3.8/site-packages/easybuild/base/rest.py", line 215, in get_connection
    connection = self.opener.open(request)
  File "/usr/lib64/python3.8/urllib/request.py", line 531, in open
    response = meth(req, response)
  File "/usr/lib64/python3.8/urllib/request.py", line 640, in http_response
    response = self.parent.error(
  File "/usr/lib64/python3.8/urllib/request.py", line 569, in error
    return self._call_chain(*args)
  File "/usr/lib64/python3.8/urllib/request.py", line 502, in _call_chain
    result = func(*args)
  File "/usr/lib64/python3.8/urllib/request.py", line 649, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 405: Method Not Allowed
```

### PR DESCRIPTION
**Pull request by migueldiascosta**
fixes #3501

### CODE REVIEW

**Comment by migueldiascosta** [comment]
hm, I suppose this will break tests that use merged PRs...

**Comment by boegel** [comment]
@migueldiascosta Is this still WIP?

**Comment by migueldiascosta** [comment]
> @migueldiascosta Is this still WIP?

No, forgot to re-edit the title

File: easybuild/tools/github.py [hunk 1]
```diff
@@ -1331,6 +1331,9 @@ def merge_pr(pr):
     force = build_option('force')
     dry_run = build_option('dry_run') or build_option('extended_dry_run')
 
+    if pr_data['merged'] and not dry_run:
```
Comment (boegel) [inline easybuild/tools/github.py:1334]: Should we also take into account closed PRs?

Also, why doesn't this make the `merge_pr` test in `test/framework/options.py` fail?

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/tools/github.py (hunk 1 above)
Comment (migueldiascosta) [reply easybuild/tools/github.py:1334]: > Should we also take into account closed PRs?

will do

> Also, why doesn't this make the `merge_pr` test in `test/framework/options.py` fail?

that's why I added the `not dry_run`

Comment (boegel) [reply easybuild/tools/github.py]: Should we update the test so it tries `--merge-pr` without dry run on a merged/closed PR to cover this change in the tests?

**Review by boegel — APPROVED** [review]
lgtm

**Pull request merged**